resources/party.ts

```diff
@@ -10,7 +19,7 @@ export default class PartyTracker {
   allianceIds_: string[] = [];
   nameToRole_: Record<string, Role> = {};
   idToName_: Record<string, string> = {};
-  roleToPartyNames_: Record<Role, string[]> = {} as Record<Role, string[]>;
+  roleToPartyNames_: Record<Role, string[]> = emptyRoleToPartyNames;
```

Comment:
Won't this object be shared cross `PartyTracker` instance?

Reply:
Oh whoops, thanks!

Reply:
I'm not sure if it works after your changes...

Copy and original object share array, too.

Reply:
Thanks for catching my (repeated) mistakes.  *<_<*